### PR TITLE
fee_adjustments: add integration testing

### LINKED ISSUE #173: Move Fee Adjustment Logic into new pallet
After what was done in #161 and what was described [here](https://github.com/thrumdev/blobs/pull/161#issuecomment-1871457847), all the Fee Adjustment Logic should be refactored into a new pallet.

This refactor would then require a small change in the `runtime/lib.rs` to start using the new fee adjustment logic contained in the pallet

### PR DESCRIPTION
**Pull request by gabriele-0201**
closes #173

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/176?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #174** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/174?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #175** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/175?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #176** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/176?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/176?utm_source=stack-comment).

File: sugondat/chain/runtimes/sugondat-kusama/tests/fee_adjustments.rs [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+        // initial value configured on module
+        let mut fm = Multiplier::one();
+        assert_eq!(fm, TransactionPayment::next_fee_multiplier());
+
+        let mut iterations: u64 = 0;
+        loop {
+            let next = MultiplierType::Fee.runtime_multiplier_update(fm);
+            // if no change, panic. This should never happen in this case.
+            if fm == next {
+                panic!("The fee should ever increase");
```
Comment (rphmeier) [inline sugondat/chain/runtimes/sugondat-kusama/tests/fee_adjustments.rs:336]: never?

**Review by rphmeier — APPROVED** [review]

File: sugondat/chain/runtimes/sugondat-kusama/tests/fee_adjustments.rs (hunk 1 above)
Comment (gabriele-0201) [reply sugondat/chain/runtimes/sugondat-kusama/tests/fee_adjustments.rs:336]: Yes, I also think it should be always, but I found that comment in the polkadot-sdk repo so I thought It was just me not knowing English enough 😅

I will correct it!

**Comment by rphmeier** [comment]
### Merge activity

* **Jan 18, 7:43 PM**: @rphmeier started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/176).

**Pull request merged**